### PR TITLE
[ONNX] Numpy style binary broadcasting

### PR DESCRIPTION
**Pull request by tsocha**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/utils/broadcasting.cpp [hunk 1]
```diff
@@ -17,12 +17,85 @@
...
+#include "ngraph/op/broadcast.hpp"
+#include "ngraph/op/reshape.hpp"
+
 #include "broadcasting.hpp"
 
 namespace ngraph
 {
     namespace onnx_import
     {
+        std::vector<Shape> calculate_numpy_broadcast_shape(Shape left_shape, Shape right_shape)
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/utils/broadcasting.cpp]: Please add header for `Shape` class.

Comment (arogowie-intel) [reply src/ngraph/frontend/onnx_import/utils/broadcasting.cpp]: If this is only a helper function pleas make it `static` and outside `ngraph` namespaces scope.

Comment (arogowie-intel) [reply src/ngraph/frontend/onnx_import/utils/broadcasting.cpp]: Pass arguments as const references and create function local copies where needed. You may name them `extended_left_shape` like.

File: src/ngraph/frontend/onnx_import/utils/broadcasting.hpp [hunk 2]
```diff
@@ -25,6 +25,14 @@ namespace ngraph
 {
     namespace onnx_import
     {
+        NodeVector binary_op_numpy_broadcasting(const std::shared_ptr<ngraph::Node>& left,
+                                                const std::shared_ptr<ngraph::Node>& right);
+
+        inline NodeVector binary_op_numpy_broadcasting(NodeVector inputs)
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/utils/broadcasting.hpp]: Please add documentation.

File: src/ngraph/frontend/onnx_import/utils/broadcasting.cpp [hunk 3]
```diff
@@ -17,12 +17,86 @@
 #include <numeric>
 #include <vector>
 
+#include "ngraph/op/broadcast.hpp"
+#include "ngraph/op/reshape.hpp"
+
 #include "broadcasting.hpp"
 
+static std::vector<ngraph::Shape> calculate_numpy_broadcast_shape(ngraph::Shape left_shape,
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/utils/broadcasting.cpp]: Add some documentation. What is this function supposed to do?
You can provide a link to numpy documentation.

File: src/ngraph/frontend/onnx_import/utils/broadcasting.hpp [hunk 4]
```diff
@@ -25,6 +25,25 @@ namespace ngraph
 {
     namespace onnx_import
     {
+        /// \brief Cast shape of two nodes to make them compatible for an element-wise binary operation.
+        ///
+        /// \param left Node which contain input of binary op.
+        /// \param right Node which contain input of binary op.
+        ///
+        /// \return Left and right node after broadcasting.
+        NodeVector binary_op_numpy_broadcasting(const std::shared_ptr<ngraph::Node>& left,
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/utils/broadcasting.hpp]: Change name to begin with the imperative mood `broadcast_...`

File: src/ngraph/frontend/onnx_import/utils/broadcasting.hpp [hunk 5]
```diff
@@ -25,6 +25,25 @@ namespace ngraph
...
+                                                const std::shared_ptr<ngraph::Node>& right);
+
+        /// \brief Cast shape of two nodes to make them compatible for an element-wise binary operation.
+        ///
+        /// \param ng_inputs Left and right node (inputs of the binary op).
+        ///
+        /// \return Left and right node after broadcasting.
+        inline NodeVector binary_op_numpy_broadcasting(NodeVector inputs)
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/utils/broadcasting.hpp]: Change name as above.

File: src/ngraph/frontend/onnx_import/utils/broadcasting.cpp [hunk 6]
```diff
@@ -17,12 +17,86 @@
...
+            auto numpy_shapes = calculate_numpy_broadcast_shape(left_shape, right_shape);
+            auto output_shape = numpy_shapes.at(0);
+            auto left_full_shape = numpy_shapes.at(1);
+            auto right_full_shape = numpy_shapes.at(2);
+
+            std::vector<size_t> left_ones_positions;
+            std::vector<size_t> right_ones_positions;
+            Shape new_left_shape;
+            Shape new_right_shape;
+            for (auto index = 0; index < output_shape.size(); ++index)
```
Comment (postrational) [inline src/ngraph/frontend/onnx_import/utils/broadcasting.cpp]: Add some comments. What are you aiming to do here. What values do you want to produce?

**Pull request merged**